### PR TITLE
Joystick Bugfixes

### PR DESCRIPTION
**Pull request by mojo-hakase**
Made two bugfixes concerning Joystick support
When enabling Joystick support in GUI while running, no Joysticks are found, restart was needed.
Disabling Joystick support via jsonrpc caused kodi to crash immediately (SegFault), because the CJoystick class is not thread safe.

### CODE REVIEW

File: xbmc/input/InputManager.h [hunk 1]
```diff
@@ -268,4 +268,5 @@ class CInputManager : public ISettingCallback
 #if defined(HAS_SDL_JOYSTICK) 
   CJoystick m_Joystick;
 #endif
+  int m_ShallSetJoystickEnabled;
```
Comment (Jalle19) [inline xbmc/input/InputManager.h]: Couldn't this be a boolean?

Comment (mojopa19) [inline xbmc/input/InputManager.h]: 3 states are needed:
0: nothing changed
1: Joystick support enabled
-1: Joystick support disabled

Comment (garbear) [inline xbmc/input/InputManager.h]: if using multiple states, use an enum or enum class

**Comment by garbear** [comment]
This code is being removed in the near future. All joystick code is getting ripped out and replaced with a binary add-on, [peripheral.joystick](https://github.com/kodi-game/peripheral.joystick).

Some of the issues could have made the leap to the new code. In particular, the SDL stuff now lives at https://github.com/kodi-game/peripheral.joystick/tree/master/src/api/sdl. Could you see if these commits still apply to the new code? If so, we'll move the discussion over there.

**Pull request closed without merging**